easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeGNU-22.12-rocm-5.2.3.eb

```diff
@@ -0,0 +1,86 @@
+easyblock = 'CMakeMake'
+
+name =    'AdaptiveCpp'
+version = '23.10.0'
+versionsuffix = '-rocm-5.2.3'
+
+homepage = 'https://adaptivecpp.github.io/'
+
+whatis = [
+    "Description: AdaptiveCPP is a modern SYCL implementation targeting CPUs and GPUs, with a focus on leveraging existing toolchains such as CUDA or HIP"
+]
+
+description = """
+hipSYCL is a modern SYCL implementation targeting CPUs and
+GPUs, with a focus on leveraging existing toolchains such as CUDA or HIP.
+
+It targets any CPU via OpenMP, NVIDIA GPUs via CUDA, AMD GPUs via HIP/ROCm and 
+Intel GPUs via oneAPI Level Zero and SPIR-V (experimental). However, this 
+module is for LUMI and does not contain the NVIDIA and Intel GPU support.
+"""
+
+toolchain = {'name': 'cpeGNU', 'version': '22.12'}
+toolchainopts = {'verbose': False, 'openmp': True}
+
+#source_urls = ["https://github.com/AdaptiveCpp/AdaptiveCpp/archive/refs/tags/"]
+#sources =     ["v%(version)s.tar.gz"]
+
+sources = [{
+    'filename': '%(name)s-%(version)s.tar.gz',
+    'git_config': {
+        'url': 'https://github.com/AdaptiveCpp',
+        'repo_name': '%(name)s',
+        'commit': 'v%(version)s',
+        'keep_git_dir': True,
+    }
+}]
+
+dependencies = [
+    ('cray-python/3.9.13.1', EXTERNAL_MODULE),
+    ('rocm/5.2.3', EXTERNAL_MODULE),
+    ('Boost', '1.81.0'),
+]
+
+builddependencies = [
+    ('buildtools', '%(toolchain_version)s', '', True),
+    ('libxml2', '2.9.14'),
+]
+
+preconfigopts = 'export PATH=${ROCM_PATH}/llvm/bin:$PATH && '
+
+configopts  = '-DROCM_PATH=$ROCM_PATH '
+configopts += '-DCMAKE_C_COMPILER=cc '
+configopts += '-DCMAKE_CXX_COMPILER=CC '
+configopts += '-DWITH_ACCELERATED_CPU=ON '
+configopts += '-DWITH_CPU_BACKEND=ON '
+configopts += '-DWITH_CUDA_BACKEND=OFF '
+configopts += '-DWITH_ROCM_BACKEND=ON '
+configopts += '-DWITH_OPENCL_BACKEND=OFF '
+configopts += '-DWITH_LEVEL_ZERO_BACKEND=OFF '
+configopts += '-DDEFAULT_GPU_ARCH=gfx90a '
+configopts += '-DAMDGPU_TARGETS="gfx90a" '
+configopts += '-DGPU_TARGETS="gfx90a" ' 
+configopts += '-DBoost_NO_BOOST_CMAKE=TRUE '
+configopts += '-DBoost_NO_SYSTEM_PATHS=TRUE '
+configopts += '-DWITH_SSCP_COMPILER=OFF ' 
+configopts += '-DROCM_CXX_FLAGS="--gcc-toolchain=$GCC_PATH/snos/" '
+configopts += '-DLLVM_DIR=${ROCM_PATH}/llvm/lib/cmake/llvm/ '
```

Comment:
As documented in the AdaptiveCpp install documentation, building against ROCm LLVM is not recommended and might break support for most of our more compiler-focused features, including the single-pass compiler. This is because ROCm LLVM does not expose a stable API as it typically sits "in between" two LLVM versions, and thus can only be supported on a "may or may not work"-basis. If there is a way to target an official LLVM release instead, that would be preferred.

Reply:
As discussed elsewhere, we test with and primarily recommend ROCm LLVM for GROMACS (which is the main use-case for ACPP on LUMI), so unless someone wants to provide the effort to also test with upstream LLVM, I think this is the GROMACS dependency that we need first and foremost.